api/src/server.rs

```diff
@@ -307,6 +307,27 @@ where
 
         let event_id = decode_event_id(&id)?;
         let event_data = decode_event_data(&data)?;
+
+        if let Some(network_id) = event_id.network_id() {
```

Comment:
nit : can we move this logic to network.rs add a method there and maybe remove multiple if statements for better readability
```
Suggestion to do something like this : 
fn process_event(id: &str, data: &str, self_network: Network) -> Result<EventsPostResponse> {
    let event_id = decode_event_id(id)?;
    let event_data = decode_event_data(data)?;
    let network_id = event_id.network_id().ok_or_else(|| anyhow!("event does not have network id"))?;
    let event_network = Network::try_from_id(network_id)?;

    if event_network != self_network {
        return Ok(EventsPostResponse::BadRequest(format!(
            "event network does not match node network: {} != {}",
            event_network, self_network
        )));
    }

    // Proceed with event processing...
}
```